### PR TITLE
Add Dockerfile with dependencies and docker-compose script

### PR DESCRIPTION
**Pull request by ewelinagr**

### CODE REVIEW

**Comment by ewelinagr** [comment]
There is a problem with the postgres image - not creating a proper role specified by POSTGRES_USER env.

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM openjdk:8-jre-alpine
+
+ENV APP_PORT 8086
+ENV WHEREABOUTS_VERSION "0.0.1-SNAPSHOT"
+# to be removed and replaced with WHEREABOUTS_VERSION after first release
+ENV WHEREABOUTS_SNAPSHOT_VERSION "0.0.1-20190606.073652-5"
+ENV WHEREABOUTS_JAR_URL https://repo.thehyve.nl/content/repositories/snapshots/nl/thehyve/whereabouts/${WHEREABOUTS_VERSION}/whereabouts-${WHEREABOUTS_SNAPSHOT_VERSION}.jar
```
Comment (gijskant) [inline docker/Dockerfile]: This also works: https://repo.thehyve.nl/service/local/artifact/maven/redirect?r=snapshots&g=nl.thehyve&a=whereabouts&v=0.0.1-SNAPSHOT&p=jar

Comment (gijskant) [reply docker/Dockerfile]: Will get you the latest snapshot.

Comment (ewelinagr) [reply docker/Dockerfile]: great! thanks, I was looking for this possibility, but couldn't find a proper pattern for the url

Comment (ewelinagr) [reply docker/Dockerfile]: actually this gives me: "307 Temporary Redirect"

Comment (gijskant) [reply docker/Dockerfile]: Weird, the `wget` in BusyBox apparently does not handle redirects.
This works:
```docker
RUN adduser -h "${APP_USR_HOME}" -D ${APP_USR} && \
    sed -i 's/\r//' /entrypoint.sh && \
    chown "${APP_USR}" /entrypoint.sh && \
    chmod u+x /entrypoint.sh && \
    # download the jar
    apk add curl && \
    curl -L "${WHEREABOUTS_JAR_URL}" -o "${APP_USR_HOME}/whereabouts-${WHEREABOUTS_VERSION}.jar"
```

Comment (ewelinagr) [reply docker/Dockerfile]: nice! it works indeed! I updated the PR.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+services:
+  whereabouts-database:
+    container_name: whereabouts-database
+    image: postgres:11-alpine
+    environment:
+      POSTGRES_USER: wa
+      POSTGRES_PASSWORD: wa
+      POSTGRES_DB: whereabouts
+    volumes:
+      - db-data:/var/lib/postgresql/data
```
Comment (gijskant) [inline docker/docker-compose.yml]: I have the same issue with PostgreSQL not having role `wa`. I think is has to do with the name of the volume, probably should be a more unique name, probably collides with the gb-backend one.
See `docker volume ls` for the list of volumes.

Comment (ewelinagr) [reply docker/docker-compose.yml]: you're right, after changing the name of the volume and removing the previous one it works!

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+FROM openjdk:8-jre-alpine
+
+ENV APP_PORT 8086
```
Comment (gijskant) [inline docker/Dockerfile]: This is the default port now, right? It can also be updated in the README.md then.

Comment (gijskant) [reply docker/Dockerfile]: Hm, apparently that's not what server.port in application.yml or application-default.yml does.

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -e
+
+cat > "${APP_CONFIG_FILE}" <<EndOfMessage
+---
+keycloak:
```
Comment (gijskant) [inline docker/entrypoint.sh]: Okay, Spring configuration overriding/merging is a bit weird. Apparently we need to add all keycloak settings here, so also add:
```yaml
  bearer-only: true
  use-resource-role-mappings: true
```

Comment (gijskant) [reply docker/entrypoint.sh]: Looks like all of the internal configuration is ignored when using externalised config. I don't understand why. https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config

Comment (gijskant) [reply docker/entrypoint.sh]: Okay, finally found something that works fine:
```
exec java -jar -server \
        "-Djava.awt.headless=true" \
        "-Dmail.mime.decodeparameters=true" \
        "-Djava.security.egd=file:///dev/urandom" \
        "-Dspring.config.location=classpath:/application.yml,${APP_CONFIG_FILE}" \
    "whereabouts-${WHEREABOUTS_VERSION}.jar"


```

Comment (ewelinagr) [reply docker/entrypoint.sh]: thanks a lot! I didn't find any other solution yet... Btw, do we need other options for this command, like "-Djava.awt.headless=true" or "-Djava.awt.headless=true"?

Comment (gijskant) [reply docker/entrypoint.sh]: But also some slight changes are required in the internal config, e.g., the `bearer-only` and `use-resource-role-mappings` need to be moved to `application.yml`.

Comment (gijskant) [reply docker/entrypoint.sh]: With #12 merged and `"-Dspring.config.location=classpath:/config/application.yml,${APP_CONFIG_FILE}"` added it should work.

Comment (ewelinagr) [reply docker/entrypoint.sh]: yes, it works for me, I will push the "Dspring.config" change

**Pull request merged**